### ISSUE TITLE
Version-projected sub-namespaces are losing their parent

### ISSUE THREAD

**Issue by daviwil** [microsoft/cadl#811]
While working on https://github.com/Azure/cadl-azure/pull/1754 I noticed that sub-namespaces like `Microsoft.Contoso` are losing their parent (i.e. `Contoso` no longer has parent `Microsoft`) after being projected for versioning.  The consequence of this issue is that the `getNamespaceString` function only returns the name of the sub-namespace "Contoso" and not the full namespace "Microsoft.Contoso".

You can see an example of this issue here: [Playground Link](https://cadlplayground.z22.web.core.windows.net/cadl-azure/prs/1754/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwrTGnZlcnNpb25pbmfMIGF6dXJlLXRvb2xzL8UscHJvdmlkZXJodWLdKMYRY29yZd8nxBFyZXNvdXJjZS1tYW5hZ2VyIjsKCkBhcm1OYW1lc3BhY2UKQHNlcnZpY2VUaXRsZSgiQ29udG9zb1DnAI1IdWJDbGllbnQiKckqVuYA0CgiMjAyMS0xMC0wMS1wcmV2aWV3xCZkb2PJRyBS5wCAIMhRIOYAiW1lbnQgQVBJLsQy5wEhZWREZXBlbmRlbmN5KEHkAL8uyD9NxTVyLsd7cy52MV8wX1DGdV8xKQpu6ADQIE1pY3Jvc29mdC7yAMs7Cgp1c2luZyBDYWRsLkh0dHA7zBFSZXN0zRHHb2luZ8gX5gCXQ29yZc4S7wCp5AFm5QEIQSDyAIYg6AGVIikKbW9kZWwgRW1wbG95ZWUgaXMgVHJhY2tlZMhRPMgcUHJvcGVydGllcz4gewogIMZg5AHIIG9mIGXHJiIpxBtmb3JtYXQoIl5bYS16QS1aMC05LV17MywyNH0kxiJrZXkoIsg1xEXGF3BhdGjECHNlZ%2BQBosojc8Ug5AFmOiBzdHLlARJ96ADl6QC%2FcOkApPEA2%2BoAv%2BsAvkFn8gC9YWdlPzogaW50MzI7CskqQ2l0edErY2l0eT%2FqAJDJLVByb2ZpbNJb6QEYYmFzZTY0VXJsxRdwxi8%2FOiBieXRlc8ZJdmlzaWJpbGl0eSgicmVhZMZBxV9UaGUgc3RhdHVzxGJ0aGUgbGFzdCDkAOBhdGlvbuQC1MVWxETlAj5TdGF0xGDlAgXMFOYBRWFybegB70%2FIRXMKaW50ZXJm5ALT6AFEcyBleHRlbmRz6QNXyi%2FpAigs8wFzPiB7xGhrbm93blZhbHVlcyjxAItLVugBs9Eb5AKX6QFlQGxyb8QWdXMKZW51bdIuS1blAeIuLukDFdMjLAogzTrEEFVwZGF0xwxEZWxlyAxBY2NlcHRlZCwKfQo%3D)

The Cadl file compiles successfully, but take a look at the routes in the OpenAPI output:

```json
  "paths": {
    "/providers/ContosoProviderHub/operations": {
      ...
    },
    "/subscriptions/{subscriptionId}/providers/ContosoProviderHub/employees": {
      ...
    },
   "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/ContosoProviderHub/employees": {
      ...
    },
```

You can see that the provider namespace shows up as `Contoso` in the route paths instead of `Microsoft.Contoso`.

Is it possible this namespace loss issue was caused by PR https://github.com/microsoft/cadl/pull/792?

**Comment by markcowl** [comment]
pri: 0

**Comment by daviwil** [comment]
Fixed by #814.
